dev/tests/api-functional/testsuite/Magento/GraphQl/SendFriend/SendFriendTest.php

```diff
@@ -0,0 +1,469 @@
+                  email:"recipient1@mail.com"
+               },
+              {
+                  name: "Recipient Name 2"
+                  email:"recipient2@mail.com"
+              }
+          ]
+        } 
+    ) {
+        sender {
+            name
+            email
+            message
+        }
+        recipients {
+            name
+            email
+        }
+    }
+}
+QUERY;
+        $this->expectException(\Exception::class);
+        $this->expectExceptionMessage('Please provide Email of sender.');
+        $this->graphQlQuery($query);
+    }
+
+    /**
+     * @magentoApiDataFixture Magento/SendFriend/_files/product_simple.php
+     */
+    public function testSendWithoutSenderMessage()
+    {
+        $query =
+            <<<QUERY
+mutation {
+    sendEmailToFriend(
+        input: {
+          product_id: 1	
+          sender: {
+            name: "Name"
+            email: "e@mail.com"
+            message: ""
+        }          
+          recipients: [
+              {
+                  name: "Recipient Name 1"
+                  email:"recipient1@mail.com"
+               },
+              {
+                  name: "Recipient Name 2"
+                  email:"recipient2@mail.com"
+              }
+          ]
+        } 
+    ) {
+        sender {
+            name
+            email
+            message
+        }
+        recipients {
+            name
+            email
+        }
+    }
+}
+QUERY;
+        $this->expectException(\Exception::class);
+        $this->expectExceptionMessage('Please provide Message.');
+        $this->graphQlQuery($query);
+    }
+
+    /**
+     * @magentoApiDataFixture Magento/SendFriend/_files/product_simple.php
+     * @magentoApiDataFixture Magento/SendFriend/Fixtures/sendfriend_configuration.php
+     */
+    public function testLimitMessagesPerHour()
+    {
+
+        /** @var SendFriend $sendFriend */
+        $sendFriend = $this->sendFriendFactory->create();
+
+        $query =
+            <<<QUERY
+mutation {
+    sendEmailToFriend(
+        input: {
+          product_id: 1
+          sender: {
+            name: "Name"
+            email: "e@mail.com"
+            message: "Lorem Ipsum"
+        }
+          recipients: [
+                {
+                  name: "Recipient Name 1"
+                  email:"recipient1@mail.com"
+               },
+               {
+                  name: "Recipient Name 2"
+                  email:"recipient2@mail.com"
+               }
+
+          ]
+        }
+    ) {
+        sender {
+            name
+            email
+            message
+        }
+        recipients {
+            name
+            email
+        }
+    }
+}
+QUERY;
+        $this->expectException(\Exception::class);
+        $this->expectExceptionMessage("You can't send messages more than {$sendFriend->getMaxSendsToFriend()} times an hour.");
+
+        for ($i = 0; $i <= $sendFriend->getMaxSendsToFriend() + 1; $i++) {
```

Comment:
So, if value will be `10000` we will have huge cycle in tests

Reply:
It is not clear so far how this ```10000``` can appear :).